apis/flowcollector/v1beta2/flowcollector_types.go

```diff
@@ -373,6 +373,80 @@ type FlowCollectorIPFIXReceiver struct {
 	Transport string `json:"transport,omitempty"`
 }
 
+type FlowCollectorOpenTelemetryLogs struct {
+	// Set `enable` to `true` to send logs to Open Telemetry receiver.
+	//+kubebuilder:default:=true
+	Enable *bool `json:"enable,omitempty"`
+}
+
+type FlowCollectorOpenTelemetryMetrics struct {
+	// Set `enable` to `true` to send metrics to Open Telemetry receiver.
+	//+kubebuilder:default:=true
+	Enable *bool `json:"enable,omitempty"`
+
+	// How often should metrics be sent to collector
+	// +kubebuilder:default:="20s"
+	PushTimeInterval *metav1.Duration `json:"pushTimeInterval,omitempty"`
+}
+
+type FlowCollectorOpenTelemetryTraces struct {
+	// Set `enable` to `true` to send traces to Open Telemetry receiver.
+	//+kubebuilder:default:=true
+	Enable *bool `json:"enable,omitempty"`
+
+	// Separate span for each prefix listed
+	//+optional
+	SpanSplitter []string `json:"spanSplitter,omitempty"`
+}
+
+type GenericTransformRule struct {
+	Input      string `json:"input,omitempty"`
+	Output     string `json:"output,omitempty"`
+	Multiplier int    `json:"multiplier,omitempty"`
+}
+
+type GenericTransform []GenericTransformRule
+
+type FlowCollectorOpenTelemetry struct {
+	// Address of the Open Telemetry receiver
+	// +kubebuilder:default:=""
+	TargetHost string `json:"targetHost"`
+
+	// Port for the Open Telemetry receiver
+	TargetPort int `json:"targetPort"`
+
+	// Protocol of Open Telemetry connection. The available options are `http` and `grpc`.
+	// +optional
```

Comment:
it should be an enum I guess

Reply:
sure, https://github.com/netobserv/network-observability-operator/pull/671/commits/efdb0d4e59cf584864ef8acc96cdf6fdf2b3f8d7